File: 24-Template-Method/src/go/tea.go
package main

import "fmt"

// Tea fills in the varying steps; it embeds DefaultBeverage for the hook.
type Tea struct {
	DefaultBeverage
}

func (Tea) Brew()          { fmt.Println("  Steeping the tea") }
func (Tea) AddCondiments() { fmt.Println("  Adding lemon") }

File: 24-Template-Method/src/rust/tea.rs
use crate::beverage::CaffeineBeverage;

/// Fills in the varying steps for tea.
pub struct Tea;

impl CaffeineBeverage for Tea {
    fn brew(&self) {
        println!("  Steeping the tea");
    }

    fn add_condiments(&self) {
        println!("  Adding lemon");
    }
}
